sw/device/lib/dif/autogen/BUILD

```diff
@@ -0,0 +1,19 @@
+# Copyright lowRISC contributors.
```

Comment:
We maybe should eliminate this build file and use the paths to the autogenerated resources directly in their corresponding DIF libarary/test definitions (eliminating this BUILD file will prevent bazel from treating this subdir as a subpackage and will let you use paths to the files directly, as in the example below).

```
cc_library(
    name = "uart",
    srcs = [
        "dif_uart.c",
        "autogen/dif_uart_autogen.inc",
        "autogen/dif_uart_autogen.c",
  ],
    hdrs = ["dif_uart.h"],
    deps = [
        "base",
        "//hw/ip/uart/data:uart_regs",
        "//sw/device/lib/base",
    ],
)

cc_test(
    name = "uart_unittest",
    srcs = [
        "dif_uart_unittest.cc",
        "dif_uart_unittest.h",
        "autogen/dif_uart_autogen_unittest.c",
        ...,      
    ],
    defines = [
        "MOCK_MMIO=1",
    ],
    deps = [
        "base",
        "//hw/ip/uart/data:uart_regs",
        "//sw/device/lib/base",
        "//sw/device/lib/base/testing",
        "//sw/device/lib/base/testing:mock_mmio",
        "@googletest//:gtest_main",
    ],
)
```

I spoke to ttripel@ to confirm my suspicions after reading the autogen/uart code:  The autogen code isn't meant to stand alone as its own library - it's meant to be part of the UART DIF.  In that case, we should just include it into the original dif_uart rule.

Reply:
I did not know you can remove it to handle subdirectories that way, and I'm still playing catchup on the autogen code. You're right, this is better. I'll add running the autogeneration to the Bazel burndown as a thing to consider in the future.